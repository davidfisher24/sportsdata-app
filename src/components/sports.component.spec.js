import { render, cleanup } from '../test-utils'
import { Provider } from 'react-redux'
import { BrowserRouter } from "react-router-dom";
import SportsList from './sports.component'
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('Sports List Component', () => {
  let store; 

  afterEach(() => {
    cleanup() 
  })

  it('matches snapshot', () => {
    store = mockStore({ 
        sports: {
          data: [
            {id: 1, desc: 'Football'},
            {id: 2, desc: 'Tennis'},
          ],
          error: null
        }
    });
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
            <SportsList />
        </BrowserRouter>
      </Provider>
    );
    expect(component.container).toMatchSnapshot()
  });

  it('renders error on error data', () => {
    store = mockStore({ 
        sports: {
          data: null,
          error: new Error
        }
    });
    const { getByTestId } = render(
      <Provider store={store}>
        <BrowserRouter>
            <SportsList />
        </BrowserRouter>
      </Provider>
    );
    expect(getByTestId("error-container")).toBeTruthy();
  });
});