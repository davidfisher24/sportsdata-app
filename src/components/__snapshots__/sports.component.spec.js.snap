// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sports List Component matches snapshot 1`] = `
<div>
  <div>
    <div
      class="list row"
    >
      <div
        class="col-md-8"
      >
        <h1
          class="h4"
        >
          Sports List
        </h1>
        <ul
          class="list-group"
        >
          <a
            href="/sports/1"
          >
            <li
              class="list-group-item d-flex justify-content-between align-items-center"
            >
              Football
            </li>
          </a>
          <a
            href="/sports/2"
          >
            <li
              class="list-group-item d-flex justify-content-between align-items-center"
            >
              Tennis
            </li>
          </a>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
